src/lib/boavizta/index.ts

```diff
@@ -164,219 +55,178 @@ export class BoaviztaCpuOutputModel
-    }
-  }
+    delete staticParams.verbose;
 
-  async validateInstanceType(staticParamsCast: object) {
-    if (!('provider' in staticParamsCast)) {
-      throw new InputValidationError(
-        this.errorBuilder({
-          message: "Missing 'provider' parameter from configuration",
-        })
-      );
-    }
+    await this.validateProvider(safeStaticParams);
+    await this.validateInstanceType(safeStaticParams);
+    await this.validateLocation(safeStaticParams);
 
-    if (!('instance-type' in staticParamsCast)) {
-      throw new InputValidationError(
-        this.errorBuilder({
-          message: "Missing 'instance-type' parameter from configuration",
-        })
-      );
+    if ('expected-lifespan' in safeStaticParams) {
+      this.expectedLifespan = safeStaticParams['expected-lifespan'] as number;
     }
 
-    const provider = staticParamsCast.provider as string;
-
-    if (
-      this.instanceTypes[provider] === undefined ||
-      this.instanceTypes[provider].length === 0
-    ) {
-      this.instanceTypes[provider] =
-        await this.supportedInstancesList(provider);
-    }
+    this.sharedParams = Object.assign({}, staticParams);
 
-    if ('instance-type' in staticParamsCast) {
-      if (
-        !this.instanceTypes[provider].includes(
-          staticParamsCast['instance-type'] as string
-        )
-      ) {
-        throw new UnsupportedValueError(
-          this.errorBuilder({
-            message: `Invalid 'instance-type' parameter: '${
-              staticParamsCast['instance-type']
-            }'. Valid values are : ${this.instanceTypes[provider].join(', ')}`,
-          })
-        );
-      }
-    }
+    return this.sharedParams;
   }
 
-  async validateProvider(staticParamsCast: object) {
-    if (!('provider' in staticParamsCast)) {
+  /**
+   * Fetches data from the Boavizta API for the Cloud model.
+   */
+  protected async fetchData(usageData: object | undefined): Promise<object> {
+    if (this.sharedParams === undefined) {
       throw new InputValidationError(
         this.errorBuilder({
-          message: "Missing 'provider' parameter from configuration",
+          message: 'Missing configuration parameters',
         })
       );
-    } else {
-      const supportedProviders = await this.supportedProvidersList();
-
-      if (!supportedProviders.includes(staticParamsCast.provider as string)) {
-        throw new InputValidationError(
-          this.errorBuilder({
-            message: `Invalid 'provider' parameter '${
-              staticParamsCast.provider
-            }'. Valid values are ${supportedProviders.join(', ')}`,
-          })
-        );
-      }
     }
-  }
 
-  async supportedInstancesList(provider: string) {
-    const instances = await axios.get(
-      `https://api.boavizta.org/v1/cloud/instance/all_instances?provider=${provider}`
-    );
+    const dataCast: object = Object.assign({}, this.sharedParams, {
+      usage: usageData,
```

Comment:
usage and usage data can share the same name

Reply:
done